service/matching/workers/registry.go

```diff
@@ -0,0 +1,228 @@
+package heartbeats
+
+import (
+	"container/list"
+	"hash/fnv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	workerpb "go.temporal.io/api/worker/v1"
+)
+
+// entry wraps a WorkerHeartbeat along with its namespace and eviction metadata.
+type entry struct {
+	namespace string
```

Comment:
maybe we should use namespace id instead? in any case please use the types `namespace.Name` or `namespace.ID` to avoid confusion